scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -0,0 +1,385 @@
+        env (map (fn [[k v]]
+                      (let [env (V1EnvVar.)]
+                        (.setName env k)
+                        (.setValue env v)
+                        env))
+                 (:environment command))
+        resources (V1ResourceRequirements.)]
+    ; metadata
+    (.setName metadata (str task-id))
+
+    ; container
+    (.setName container "job")
+    (.setCommand container
+                 ["/bin/sh" "-c" (:value command)])
+
+    (.setEnv container (into [] env))
+    (.setImage container image)
+
+    (.putRequestsItem resources "memory"
+                      (Quantity. (BigDecimal/valueOf ^double (* 1024 1024 mem))
+                                 Quantity$Format/DECIMAL_SI))
+    (.putLimitsItem resources "memory"
+                    (Quantity. (BigDecimal/valueOf ^double (* 1024 1024 mem))
+                               Quantity$Format/DECIMAL_SI))
+    (.putRequestsItem resources "cpu"
+                      (Quantity. (BigDecimal/valueOf ^double cpus)
+                                 Quantity$Format/DECIMAL_SI))
+    (.setResources container resources)
+
+    ; pod-spec
+    (.addContainersItem pod-spec container)
+    (.setNodeName pod-spec hostname)
+    (.setRestartPolicy pod-spec "Never")
+
+    ; pod
+    (.setMetadata pod metadata)
+    (.setSpec pod pod-spec)
+
+    pod))
+
+(defn pod->pod-state
+  [pod]
+  (when pod
+    (let [^V1PodStatus pod-status (.getStatus pod)
+          container-statuses (.getContainerStatuses pod-status)
+          job-status (first (filter (fn [c] (= "job" (.getName c)))
+                                    container-statuses))]
+      (if job-status
+        (let [^V1ContainerState state (.getState job-status)]
+          (cond
+            (.getWaiting state)
+            {:state :pod/waiting
+             :reason (-> state .getWaiting .getReason)}
+            (.getRunning state)
+            {:state :pod/running
+             :reason "Running"}
+            (.getTerminated state)
+            (let [exit-code (-> state .getTerminated .getExitCode)]
+              (if (= 0 exit-code)
+                {:state :pod/succeeded
+                 :exit exit-code
+                 :reason (-> state .getTerminated .getReason)}
+                {:state :pod/failed
+                 :exit exit-code
+                 :reason (-> state .getTerminated .getReason)}))
+            :default
+            {:state :pod/unknown
+             :reason "Unknown"}))
+
+        {:state :pod/waiting
+         :reason "Pending"}))))
+
+
+(defn pod-state->task-status
+  [pod {:keys [state reason]}]
+  (let [task-state (case state
+                     :pod/waiting :task-staging
+                     :pod/running :task-running
+                     :pod/succeeded :task-finished
+                     :pod/failed :task-failed)
+        task-id (-> pod .getMetadata .getName)]
+
+    {:task-id {:value task-id}
+     :reason reason ; TODO(pschorf): Map as best as possible to mesos reasons
```

Comment:
Same question - is this for this PR or later?

Reply:
Same answer :)